airflow/www/extensions/init_appbuilder.py

```diff
@@ -660,15 +661,24 @@ def _process_inner_views(self):
 
 def init_appbuilder(app) -> AirflowAppBuilder:
     """Init `Flask App Builder <https://flask-appbuilder.readthedocs.io/en/latest/>`__."""
+    from airflow.auth.managers.fab.security_manager.override import FabAirflowSecurityManagerOverride
     from airflow.www.security import AirflowSecurityManager
 
-    security_manager_class = app.config.get("SECURITY_MANAGER_CLASS") or AirflowSecurityManager
+    sm_from_config = app.config.get("SECURITY_MANAGER_CLASS")
+    if sm_from_config:
+        if not issubclass(sm_from_config, AirflowSecurityManager):
+            raise Exception(
+                """Your CUSTOM_SECURITY_MANAGER must extend FabAirflowSecurityManagerOverride,
+                 not FAB's own security manager."""
+            )
+        if not issubclass(sm_from_config, FabAirflowSecurityManagerOverride):
+            warnings.warn(
+                "Please make your custom security manager inherit from "
+                "FabAirflowSecurityManagerOverride instead of AirflowSecurityManager.",
+                DeprecationWarning,
+            )
```

Comment:
We should move this piece of code inside `get_security_manager_override_class` of `FabAuthManager`